Link directly to the transcription comment

arfie opened the pull request:
This changes the posts on r/ToR_Archive to directly link to the transcription on the original post, and skips posts without a transcription (autoremoved).

For now I've pretty much copied the thread check from `verified_post_transcript` in ToR; which should probably be moved to tor_core.

Commit:
Link directly to the transcription comment

perryprog commented:
Interestingly, yesterday I was trying to browse mobile on tor_archive, (with voiceover) and I found it interesting how many times I had to click through just to get to the transcriptions.

perryprog commented on tor_archivist/main.py:
```diff
@@ -6,13 +6,30 @@
+    post.comments.replace_more(limit=0)
+
+    for comment in post.comments.list():
+        if 'www.reddit.com/r/TranscribersOfReddit' in comment.body:
```
I suppose that's one way of doing this, but what I've done for this sort of thing is this:
```python
    for comment in submission.comments:
        if footer in comment.body:
            print(f"https://reddit.com{comment.permalink}")
            transcribed.append(comment)
```

where footer is
```python
footer = """^^I'm&#32;a&#32;human&#32;volunteer&#32;content&#32;transcriber&#32;for&#32;Reddit!&#32;[If&#32;&#32;you'd&#32;like&#32;more&#32;information&#32;on&#32;what&#32;we&#32;do&#32;and&#32;why&#32;we&#32;do&#32;it,&#32;click&#32;here!](https://www.reddit.com/r/TranscribersOfReddit/wiki/index)"""
```

arfie replied:
This is the same footer check as is used in ToR. See also the second paragraph in the PR message - I want to merge (part of) the functionality from both bots into tor_core, to avoid duplicate code. But yeah, we only check for the link when detecting the footer.

perryprog replied:
If it's the same, then that's fine. 

We might want that to be a more in depth check, but that's for another PR.

arfie replied:
No, I think it's completely fine as it is now - the only way it could fail is if someone else links to our wiki in a top-level comment on the thread, and gets more upvotes than the actual transcription. Which basically would only happen if someone deliberately tries to screw up the archive, and is just as easily done if we check for the entire header. The same counted for ToR until history validation was added.

Furthermore, until the config update is done, this would make it more difficult to change the footer (for example, by adding the word "human" :P), since we'd have to change stuff in multiple places then.

perryprog replied:
Yeah, I just woke up so I'm going to blame it on that then. I take back what we said — it's good as it is.

However, it would be cool to move all of the checking things into tor_core (I think :P). That way we can use history validation in places like this.

TheLonelyGhost replied:
There's an open issue for moving things out of `tor_core` back into here for topics that only pertain to this bot. We'd have to look through each check to evaluate the best place for it to live.

Either way, as far as this thread goes, 👍

perryprog commented on tor_archivist/main.py:
```diff
@@ -39,9 +56,12 @@ def run(config):
             continue
 
+        # the original post that might have been transcribed
+        original_post = config.r.get_submission(post.url)
```
Not sure about this, but doesn't PRAW need it to be `get_submission(url=...)` when you get a submission via URL?

arfie replied:
It does work without using the kw argument, but I agree it's better style. Will change.

perryprog commented on tor_archivist/main.py:
```diff
@@ -63,11 +83,19 @@ def run(config):
+                    url=reddit_url.format(transcript.permalink))
+                logging.info('Post archived!')
+            else:
+                logging.info('Could not find the transcript - won\'t archive.')
```
We might want this to be a warn — it's a bug if we can't find the transcript on a post. Also log the URLs, and other relevant information.

arfie replied:
I suppose in some cases it should be a warning, but 99% of the time this will happen, it's because a transcript was autoremoved, in which case we just treat it like completed (see @itsthejoker's history validation PR on ToR). We already have that as a sort of warning by sending it to a separate channel in our slack. If there were an easy way to detect the other cases, that is, when something has actually gone wrong on our end, I'd definitely be in favour of making that an error/warning, but right now there isn't, and it's not really the most crucial part of the platform anyway.

perryprog replied:
Ohhh that's right. That's gonna stink.

It'd be neat if we added history validation, and transcription validation in general to tor_core, but that's not relevant to this PR.

TheLonelyGhost replied:
I'm also curious why we're avoiding archiving if it can't find the transcript. I know there's the spam filter and that it might remove a completed transcription comment, but would we want it to stick around in /r/ToR's feed afterwards?

I suppose this also lends to the idea that we might want the main tor bot to link to the detected transcription comment in /r/ToR post when detecting if the transcription is done in order to get around the deletion thing. 🤷‍♂️

perryprog reviewed (approved):
LGTM

(once you make the commit for `url=...`, but close enough)

Commit:
use submission instead of get_submission

Not sure what the difference is, but can't find doc for get_submission
and we use submission in other places. Also gave the url as KW arg for
clarity.

Commit:
update changelog

TheLonelyGhost commented:
Oops! Sorry I didn't see this earlier! Will pop back here once I have a bit more time to review in a couple of hours.

TheLonelyGhost commented on tor_archivist/main.py:
```diff
@@ -39,9 +56,12 @@ def run(config):
         # find the original post subreddit
         # take it in lowercase so the config is case insensitive
-        post_subreddit = subreddit_from_url(post.url).lower()
+        post_subreddit = original_post.subreddit
```
I like our using the `post.subreddit` attribute instead of trying to parse a URL.

Don't forget to make the subreddit name lowercase like the comments above say.

itsthejoker commented:
@TheLonelyGhost We want to avoid linking to a comment that does not exist because we're pointing people to the archive saying "everything in here should be good to go" and right now that's not even remotely true; only archiving things that have visible comments makes a lot more sense in the realm of "all of these links are accessible".

itsthejoker committed:
fix merge comflicts

Signed-off-by: Joe Kaufeld <joe.kaufeld@gmail.com>

itsthejoker committed:
update for newest version of bot

Signed-off-by: Joe Kaufeld <joe.kaufeld@gmail.com>

itsthejoker committed:
resolve comment

Signed-off-by: Joe Kaufeld <joe.kaufeld@gmail.com>
